### DISCUSSION TITLE
How can I use the agent?

### DISCUSSION THREAD

**Discussion by sukbearai** [discordjs/discord.js#9503] [Q&A]
### Which package is this bug report for?

discord.js

### Issue description

I'm using the agent, but I'm encountering an error.

<img width="1020" alt="image" src="https://user-images.githubusercontent.com/120086676/235577363-200d375a-f2d4-408d-9f55-1ccefac2e5d4.png">

### Code sample
```JavaScript
// Require the necessary discord.js classes
const { Client, Events, GatewayIntentBits } = require('discord.js');
const { token } = require('./config.json');
import { ProxyAgent } from 'undici';

// Create a new client instance
const client = new Client({ intents: [GatewayIntentBits.Guilds], rest:{ agent: new ProxyAgent(`http://127.0.0.1:7890`) } });


// When the client is ready, run this code (only once)
// We use 'c' for the event parameter to keep it separate from the already defined 'client'
client.once(Events.ClientReady, c => {
	console.log(`Ready! Logged in as ${c.user.tag}`);
});

// Log in to Discord with your client's token
client.login(token);
```
_No response_

### Versions

- discord.js 14.9.0 (`npm ls discord.js` or another package)
- Node.js 16.9.0 (`node --version`)
- TypeScript 5.0.4 (`npm ls typescript` if you use it)
- macOS Ventura 13.3.1

### Issue priority

High (immediate attention needed)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

Guilds

**Comment by MiguelSantos96** [comment]
@sukbearsuwu  Can you catch the error on the login? 

```javascript
client.login(token).catch(error => {
    console.error(`An error occurred while logging in: ${error.message}`);
});

**Comment by sukbearai** [comment]
```JavaScript
'use strict';

var undici = require('undici');

// Require the necessary discord.js classes
const { Client, Events, GatewayIntentBits } = require('discord.js');
const { token } = require('./config.json');

// Create a new client instance
const client = new Client({ intents: [GatewayIntentBits.Guilds], rest:{ agent: new undici.ProxyAgent(`http://127.0.0.1:7890`) } });


// When the client is ready, run this code (only once)
// We use 'c' for the event parameter to keep it separate from the already defined 'client'
client.once(Events.ClientReady, c => {
	console.log(`Ready! Logged in as ${c.user.tag}`);
});

// Log in to Discord with your client's token
client.login(token).catch(error => {
    console.error(`An error occurred while logging in: ${error.message}`);
});
```

<img width="817" alt="image" src="https://user-images.githubusercontent.com/120086676/235808787-c1fe5b52-8132-4f2d-80e9-16d1c337cc2f.png">

This is the proxy port.

<img width="494" alt="image" src="https://user-images.githubusercontent.com/120086676/235809266-c0fe90b0-f71d-4776-a179-baee02840fa2.png">

**Comment by Qjuh** [comment]
The ProxyAgent you pass is for the REST API connection. Your error is from establishing the WebSocket Gateway connection failing. So not related.

**Comment by MiguelSantos96** [comment]
Hi @sukbearsuwu, 

Lets try fix your client.login() issue.

How is your config.json file? Can you send it not showing your token 

it sould be like this:

```
{
    "token": "YOUR_TOKEN_HERE"
}
```

**Comment by bao-io** [comment]
I have same issue，how can i resolve this problem ? i can run my service successfully on local，but  deploy on linux with network proxy, the terminal throw above the error

**Reply by Qjuh** [reply]
You need to configure your host to correctly use the proxy in its network configuration then, not a djs issue.

**Comment by Jfs007** [comment]
I have the same problem. Has the host solved it
